libp2p/peer/peerstore.py

```diff
@@ -84,6 +86,20 @@ def peer_ids(self) -> list[ID]:
         """
         return list(self.peer_data_map.keys())
 
+    def valid_peer_ids(self) -> list[ID]:
+        """
+        :return: all of the valid peer IDs stored in peer store
+        """
+        valid_peer_ids: list[ID] = []
+        peer_ids = list(self.peer_data_map.keys())
+        for peer_id in peer_ids:
+            peer_data = self.peer_data_map[peer_id]
+            if not peer_data.is_expired():
+                valid_peer_ids.append(peer_id)
+            else:
+                peer_data.clear_addrs()
```

Comment:
```suggestion
        for peer_id, peer_data in self.peer_data_map.items():
            if not peer_data.is_expired():
                valid_peer_ids.append(peer_id)
            else:
                peer_data.clear_addrs()
```

I think we can reduce this block of code for better readability